packages/integrations/vercel/src/serverless/adapter.ts

```diff
@@ -224,6 +258,19 @@ export default function vercelServerless({
 					);
 				}
 			},
+			'astro:server:setup' ({ server }) {
+				// isr functions do not have access to search params, this middleware removes them for the dev mode
+				if (isr) {
+					const exclude_ = typeof isr === "object" ? isr.exclude ?? [] : [];
+					const exclude = exclude_.concat("/_image").map(ex => new RegExp(ex));
+					server.middlewares.use(function removeIsrParams(req, _, next) {
+						const { pathname } = new URL(`https://example.com${req.url}`);
+						if (exclude.some(ex => ex.test(pathname))) return next();
+						req.url = pathname;
+						return next();
+					})
```

Comment:
This seems good, but I worry that this will be confusing because it happens silently. Maybe log a warning?

Reply:
Yeah, I agree. Do we already have a message for when `astro dev` removes params?

Reply:
Ha, no. Turns out that we do it silently in core as well. https://github.com/withastro/astro/blob/7224809b73d2c3ec8e8aee2fa07463dc3b57a7a2/packages/astro/src/vite-plugin-astro-server/request.ts#L52

I retract my suggestion!